executor/builder.go

```diff
@@ -746,21 +753,36 @@ func (b *executorBuilder) buildExecute(v *plannercore.Execute) Executor {
 		}
 
 		if v.SnapshotTS != 0 {
-			is, err := domain.GetDomain(b.ctx).GetSnapshotInfoSchema(b.snapshotTS)
+			is, err := domain.GetDomain(b.ctx).GetSnapshotInfoSchema(v.SnapshotTS)
 			if err != nil {
 				panic(err)
 			}
 
 			if b.is.SchemaMetaVersion() != is.SchemaMetaVersion() {
 				panic(fmt.Sprintf("%d != %d", b.is.SchemaMetaVersion(), is.SchemaMetaVersion()))
 			}
+
+			ts, err := sessiontxn.GetTxnManager(b.ctx).GetReadTS()
+			if err != nil {
+				panic(e)
+			}
+
+			if v.SnapshotTS != ts {
+				panic(fmt.Sprintf("%d != %d", ts, v.SnapshotTS))
```

Comment:
For the unexpected path we could add an error log and return error here.

Reply:
This is in failpoint, panic here will not occur in product environment.